xds/internal/xdsclient/xds.go

```diff
@@ -575,13 +575,15 @@ func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy, logg
 		case *v3routepb.RouteAction_HashPolicy_Header_:
 			policy.HashPolicyType = HashPolicyTypeHeader
 			policy.HeaderName = p.GetHeader().GetHeaderName()
-			regex := p.GetHeader().GetRegexRewrite().GetPattern().GetRegex()
-			re, err := regexp.Compile(regex)
-			if err != nil {
-				return nil, fmt.Errorf("hash policy %+v contains an invalid regex %q", p, regex)
+			if p.GetHeader().GetRegexRewrite() != nil {
+				regex := p.GetHeader().GetRegexRewrite().GetPattern().GetRegex()
+				re, err := regexp.Compile(regex)
+				if err != nil {
+					return nil, fmt.Errorf("hash policy %+v contains an invalid regex %q", p, regex)
+				}
+				policy.Regex = re
```

Comment:
Is there a test checking `policy.Regex` is `nil` if `RegexRewrite` is not set in the xds response?

Reply:
Added.